File: 13-roman-to-integer/roman-to-integer.cpp
// class Solution {
// public:
//     int romanToInt(string s) {
//         unordered_map<char,int>m;
//         m['I']=1;
//         m['V']=5;
//         m['X']=10;
//         m['L']=50;
//         m['C']=100;
//         m['D']=500;
//         m['M']=1000;
//         int ans=0;
//         for(int i=0;i<s.length();i++){
//             if(m[s[i]]<m[s[i+1]]){
//                 ans=ans-m[s[i]];
//             }
//             else{
//                 ans=ans+m[s[i]];
//             }
//         }
//         return ans;
//     }
// };
class Solution{
    public:
    int romanToInt(string s){
        int i=0;
        int value=0;
        unordered_map<char,int>m;
        m['I']=1;
        m['V']=5;
        m['X']=10;
        m['L']=50;
        m['C']=100;
        m['D']=500;
        m['M']=1000;
        while(i<s.length()){
        if(i<s.length()-1&&m[s[i]]<m[s[i+1]]){
            value=value+m[s[i+1]]-m[s[i]];
            i=i+2;
        }
        else{
            value=value+m[s[i]];
            i=i+1;
        }
        }
return value;

    }
};

File: 13-roman-to-integer/roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        HashMap<Character,Integer>m=new HashMap<>();
        m.put('I',1);
        m.put('V',5);
        m.put('X',10);
        m.put('L',50);
        m.put('C',100);
        m.put('D',500);
        m.put('M',1000);
        int res=0;
        int prevValue=0;

        for(int i=s.length()-1;i>=0;i--){
            int currentValue=m.get(s.charAt(i));
            if(currentValue<prevValue){
                res-=currentValue;
            }else{
                res+=currentValue;
            }
            prevValue=currentValue;

        }
        return res;
        
    }
}